src/vault/client.clj

```diff
@@ -70,7 +79,41 @@
   (delete-secret!
     [this path]
     (swap! memory dissoc path)
-    true))
+    true)
+
+  (create-token! [this] (create-token! this nil))
+
+  (create-token!
+    [this wrap-ttl]
+    {:request_id "",
+     :lease_id "",
+     :renewable false,
+     :lease_duration 0,
+     :data nil,
+     :wrap_info
+     {:token "87e771a4-73a5-8eda-fe59-a5cc5d41a6d2",
+      :ttl 300,
+      :creation_time "2016-11-15T16:09:05.570104718-08:00",
+      :wrapped_accessor "2bde7146-78a2-541c-a073-59d6f1535186"},
+     :warnings nil,
+     :auth nil})
+
+  (unwrap!
+    [this wrap-token]
+    {:request_id "21a1a583-a03b-f465-27a9-0461979c7aa7",
```

Comment:
I would make this actually look up something from the client's internal state - perhaps placed there by `create-token!`?